prospero-common/src/main/java/org/wildfly/prospero/api/ArtifactChange.java

```diff
@@ -22,44 +22,25 @@
 
 import java.util.Optional;
 
-public class ArtifactChange {
-    private Artifact oldVersion;
-    private Artifact newVersion;
+public class ArtifactChange extends Diff {
 
     public ArtifactChange(Artifact oldVersion, Artifact newVersion) {
-        this.oldVersion = oldVersion;
-        this.newVersion = newVersion;
+        super(toGav(oldVersion == null?newVersion:oldVersion), oldVersion==null?null:oldVersion.getVersion(), newVersion==null?null:newVersion.getVersion());
     }
 
     public String getArtifactName() {
-        if (oldVersion == null) {
-            return toGav(newVersion);
-        } else {
-            return toGav(oldVersion);
-        }
+        return getName().get();
```

Comment:
Should the `getArtifactName()` return Optional too (same as the Diff)? Or should the name be non-nullable? To avoid a possible NPE inside this method...

Reply:
For the ArtifactChange, the name can never be null since it's based on the old and new Artifacts which have to include groupId and artifactId

Reply:
Right, didn't realize that. We could add something like `assert getName().isPresent() : "Artifact name should never be null.";` to make that clear for IDE inspections.